src/hdmf/backends/io.py

```diff
@@ -36,23 +36,70 @@ def read(self, **kwargs):
         return container
 
     @docval({'name': 'container', 'type': Container, 'doc': 'the Container object to write'},
-            {'name': 'exhaust_dci', 'type': bool,
-             'doc': 'exhaust DataChunkIterators one at a time. If False, exhaust them concurrently', 'default': True})
+            {'name': 'write_config', 'type': 'WriteConfig', 'doc': 'configuration settings for writing builders',
+             'default': None})
     def write(self, **kwargs):
         container = popargs('container', kwargs)
         f_builder = self.__manager.build(container, source=self.__source)
         self.write_builder(f_builder, **kwargs)
 
+    @docval({'name': 'src_io', 'type': 'HDMFIO', 'doc': 'the HDMFIO object for reading the data to export'},
+            {'name': 'container', 'type': Container,
+             'doc': ('the Container object to export. If None, then the entire contents of the HDMFIO object will be '
+                     'exported'),
+             'default': None},
+            {'name': 'read_args', 'type': dict, 'doc': 'dict of arguments to use when calling read_io.read_builder',
+             'default': dict()},
+            {'name': 'write_config', 'type': 'WriteConfig', 'doc': 'configuration settings for writing builders',
```

Comment:
'write_config' -> 'write_args'. Make it a dict. `WriteConfig` is used only internally